application/src/main/java/org/opentripplanner/model/impl/SubmodeMappingModule.java

```diff
@@ -0,0 +1,103 @@
+package org.opentripplanner.model.impl;
+
+import com.csvreader.CsvReader;
+import dagger.Module;
+import dagger.Provides;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.opentripplanner.datastore.api.DataSource;
+import org.opentripplanner.framework.application.OtpAppException;
+import org.opentripplanner.graph_builder.GraphBuilderDataSources;
+import org.opentripplanner.graph_builder.model.GraphBuilderModule;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.transit.model.basic.TransitMode;
+
+@Module
+public class SubmodeMappingModule implements GraphBuilderModule {
```

Comment:
Javadoc.

Reply:
Also, it feels a bit weird that this is both a dagger module and a GraphBuilderModule.